File: DSA-solutions/string/valid parentheses/result.java
class Solution {
    public boolean isValid(String s) {
        HashMap<Character, Character> open = new HashMap<>();
        ArrayList<Character> stack = new ArrayList<>();
        open.put('(', ')');
        open.put('[', ']');
        open.put('{', '}');
        char[] tmp = s.toCharArray();
        for (int i = 0; i < tmp.length; i++) {
            if (open.containsKey(tmp[i]))
                stack.add(open.get(tmp[i]));
            else if (stack.size() > 0 && stack.get(stack.size() - 1) == tmp[i]) {
                stack.remove(stack.size() - 1);
            } else
                return false;
        }
        return stack.size() == 0;
    }
}

File: DSA-solutions/string/valid parentheses/result.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let open = new Map([['(', ')'], ['{', '}'], ['[', ']']]);
    
    let stack = [];
    for(let ch of s){
        if(open.has(ch))
            stack.push(open.get(ch));
        else if(stack.length && stack[stack.length - 1] === ch)
            stack.pop();
        else
            return false;
    }
    
    return stack.length === 0
};